mass_mailing_unique/models/mail_mass_mailing_contact.py

```diff
@@ -0,0 +1,17 @@
+# Copyright 2018 Tecnativa - Ernesto Tejeda
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+
+
+class MailMassMailingContact(models.Model):
+    _inherit = 'mail.mass_mailing.contact'
+
+    @api.constrains('email', 'list_ids')
+    def _check_email_list_ids(self):
+        for contact in self:
```

Comment:
I'm afraid this for could lead to bottlenecks, mostly when using suggestions from below. How about using `set()`s, which allow to check for duplicates? One idea:

```python
others = self.mapped("list_ids.contact_ids") - self
if set(self.mapped("email")) & set(others.mapped("email")):
    raise ...
```

Reply:
The problem of what you propose is that when you try to create a list with two contacts with the same email address, the exception will not be launched

Reply:
Are you sure? AFAICS that's exactly what it does... It just avoids many loops to make sure the method scales fine.

Reply:
Suppose that the write method is called to modify two contacts (contact_1 and contact_2) so that both have the same email (contact@example.com) and belong to the same lists (L_1 and L_2), which had not yet been associated to no contact. In that case, the error will not be launched, because the intesection between the sets set (contact@example.com) and set () will be empty.

Reply:
Well, you're absolutely right! Thanks :wink: